sktime/contrib/critical_difference.py

```diff
@@ -0,0 +1,600 @@
+# -*- coding: utf-8 -*-
+"""Standalone function to plot critical difference diagram."""
+
+__author__ = ["Svea Meyer"]
+
+import numpy as np
+from scipy.stats import rankdata, find_repeats, distributions
+import math
+from sktime.utils.validation._dependencies import _check_soft_dependencies
+
+_check_soft_dependencies("matplotlib")
+import matplotlib.pyplot as plt
+
+
+def _check_friedman(n_strategies, n_datasets, ranked_data, alpha):
+    """
+    Check whether Friedman test is significant.
+
+    Larger parts of code copied from scipy.
+
+    Arguments
+    ---------
+    n_strategies : int
+      number of strategies to evaluate
+    n_datasets : int
+      number of datasets classified per strategy
+    ranked_data : np.array (shape: n_strategies * n_datasets)
+      rank of strategy on dataset
+
+    Returns
+    -------
+    is_significant : bool
+      Indicates whether strategies differ significantly in terms of performance
+      (according to Friedman test).
+    """
```

Comment:
I'd include a reference to which algorithm you are implementing, e.g., which formula for the approximation of the null distribution